sdk/storage/azblob/blob/models.go

```diff
@@ -146,11 +155,11 @@ func (o *DownloadToWriterAtOptions) getDownloadBlobOptions(offSet, count int64,
 	}
 }
 
-// DownloadToBufferOptions identifies options used by the DownloadToBuffer and DownloadToFile functions.
-type DownloadToBufferOptions = DownloadToWriterAtOptions
+// DownloadBufferOptions identifies options used by the DownloadBuffer and DownloadFile functions.
+type DownloadBufferOptions = downloadOptions
 
-// DownloadToFileOptions identifies options used by the DownloadToBuffer and DownloadToFile functions.
-type DownloadToFileOptions = DownloadToWriterAtOptions
+// DownloadFileOptions identifies options used by the DownloadBuffer and DownloadFile functions.
+type DownloadFileOptions = downloadOptions
```

Comment:
This and `DownloadBufferOptions` should be their own types.

Reply:
I am not clear on this one Joel. If I anonymously embed the unexported type, the fields of `downloadOptions` won't be visible, correct?

Reply:
I will say though that if we anonymously embed a common `DownloadOptions` type it does make for a clunky user experience.

Reply:
Perhaps the cleanest/most sustainable (versioning) solution is to define the public Download<Operation>Options struct with public fields with what you want and then when this is passed to the method, create an instance of the common/private struct and manually copy the fields over and then pass the common type along to the next method.

You could use that method I wrote that uses reflection to copy the same fields/types if you wanted to but this may be overkill for just these scenarios.